modules/nextflow/src/main/groovy/nextflow/extension/OpCall.groovy

```diff
@@ -126,14 +126,19 @@ class OpCall implements Callable {
     Object[] getArgs() { args }
 
     private <T> T read0(source){
+        T readChan = (T)source;
         if( source instanceof DataflowBroadcast )
-            return (T)CH.getReadChannel(source)
+            readChan = (T)CH.getReadChannel(source)
 
         if( source instanceof DataflowQueue )
-            return (T)CH.getReadChannel(source)
+            readChan = (T)CH.getReadChannel(source)
 
-        else
-            return (T)source
+        if( !ignoreDagNode )
+            // Keep track of this relationship so we can retrieve the
+            // DataflowBroadcast or DataflowQueue when rendering the DAG.
+            NodeMarker.addDataflowBroadcastPair(readChan, source)
```

Comment:
Wonder if this should also include ` source != readChan`

Reply:
I put that check here: https://github.com/mes5k/nextflow/blob/bug_1486_dag_edge_fix/modules/nextflow/src/main/groovy/nextflow/dag/DAG.groovy#L81 in case we need to use that method anywhere else.  Let me know if you'd like me to move it.

Reply:
Fine than!